### PR TITLE
[Free Trial] Improve new store sync handling.

### LINKED ISSUE #9338: [Free Trial] Investigate timeout situations
Ref: https://github.com/woocommerce/woocommerce-ios/pull/9317

There seem to be some issues where a free trial site information is not fully available after it's creation.

We should investigate possible solutions. One approach is adding the 45s delay that Android has.

### PR DESCRIPTION
**Pull request by Ecarrion**
Closes: #9338 

# Why

Currently, the app waits until a newly created site reflects the `isJetpackThePluginInstalled` and the `isJetpackConnected` properties as `true` in order to continue with the created store flow.

However, after a lot of debugging, I've noticed that those stores are not totally usable until other properties like `name`, `isWordPressComStore` or `isWooCommerceActive` are properly in sync.

These properties can take some seconds, minutes, or even days to be synced 😱 ref:  p1669597269761459/1667224844.942809-slack-CDLH4C1UZ

This PR updates the code on  `syncSites(forSiteThatMatchesSiteID: Int64)`  to not allow the creation process to continue until these properties are in sync. If the timeout expires, we will present an alert error telling the merchant that the store will be available "soon". Which is better than proceeding with a store in a faulty state.

**Note: I think we should add some analytics events before presenting the timeout alert**

# Demo

## Timeout

When the store takes too much time to sync it's better to show a timeout alert rather than allow it to continue with an uncompleted store.

https://user-images.githubusercontent.com/562080/228991703-a16dcdf0-a935-41b7-bc3a-c6922723d455.mov

## Success

Wait for all properties to be in sync before allowing the flow to complete

https://user-images.githubusercontent.com/562080/228991707-fabcf8a4-b955-44b7-9c52-73a3dd0a6117.mov

ios-tracks

# Testing Steps

Testing this is complicated as it depends on the WPCom infrastructure to sync sites at a faster or slower rate.

Basically, just try to create free trial sites and make sure that when the flow completes, all the information (store name, trial banner) is there.

If you encounter the timeout alert situation, that is fine too! 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/Store Creation/StoreCreationCoordinator.swift [hunk 1]
```diff
@@ -680,6 +684,14 @@ private extension StoreCreationCoordinator {
                     DDLogInfo("🔵 Retrying: Site available but is not a jetpack site yet...")
                     return continuation.resume(throwing: StoreCreationError.newSiteIsNotJetpackSite)
                 }
+
+                // Sometimes, as soon as the jetpack installation is done some properties like `name` and `isWordPressComStore` are outdated.
+                // In this case, let's keep retrying sites syncing. https://github.com/woocommerce/woocommerce-ios/pull/9317#issuecomment-1488035433
+                guard site.isWordPressComStore && site.isWooCommerceActive && site.name == expectedStoreName else {
+                    DDLogInfo("🔵 Retrying: Site available but properties are not yet in sync...")
+                    return continuation.resume(throwing: StoreCreationError.newSiteIsNotJetpackSite)
+                }
```
Comment (Ecarrion) [inline WooCommerce/Classes/Authentication/Store Creation/StoreCreationCoordinator.swift:693]: @jaclync not expecting a full review from you but just tagging you here in case you have some opinions on this change!

Comment (jaclync) [reply WooCommerce/Classes/Authentication/Store Creation/StoreCreationCoordinator.swift:693]: This looks good to me 👍 ideally, this can be addressed on the server side. let's hope it doesn't take more than a minute or so for the majority of the merchants 😅 

Maybe we can also track the duration in the `siteCreated` event as an event property, if it becomes an issue later on.

**Review by ThomazFB — APPROVED** [review]
Great solution, thanks for taking care of it!

> Note: I think we should add some analytics events before presenting the timeout alert

Good point. I wonder if we want to create a specific track event name for this scenario or just use the `siteCreationFailed` event with an additional parameter. Seems pretty important for us to metrify how often this problem is happening. WDYT?

**Comment by Ecarrion** [comment]
> Good point. I wonder if we want to create a specific track event name for this scenario or just use the siteCreationFailed event with an additional parameter. Seems pretty important for us to metrify how often this problem is happening. WDYT?

I think it should be a new event, as the site creation did not "fail" is just that is taking more time than expected.
